api/inference/sentiment.py

```diff
@@ -57,18 +58,41 @@ def handler(event, context):
 
     print("type: ", type(qs["subreddit"]))
     subreddit = qs["subreddit"]
-    # TODO: fetch the top 25 hottest reddit posts for the input subreddit
-    # PRAW
-    # Send a raw API request to reddit
-    print("subreddit: ", subreddit)
-    # Get most common predicted label from the top 25 headlines --> then average
-    # the predictions
-    res = nlp_pipeline(subreddit)[0]
-    print("model pred: ", res)
+    
+    # initialize praw
+    reddit = praw.Reddit(
+        client_id = 'insert_client_id',
+        client_secret = 'insert_secret_key',
+        user_agent = 'insert_user_agent'
+    )
+    
+    # Send a raw API request to reddit to fetch the 25 hottest 
+    headings = []
+    for submission in reddit.subreddit(subreddit).hot(limit=25):
+        headings.append(submission.title)
+        
+    # running model on each heading
+    model_result = []
+    for topic in headings:
```

Comment:
Feed the data as a single argument like `nlp_pipeline(headings)` instead of `for` looping. It is significantly faster this way